cmd/utils/nodecmd/chaincmd.go

```diff
@@ -232,45 +212,53 @@ func writeFailLog(key int, err error) {
 	logger.Crit(msg, "err", err)
 }
 
-func checkGenesisAndFillDefaultIfNeeded(genesis *blockchain.Genesis) *blockchain.Genesis {
-	engine := params.UseIstanbul
-	valueChanged := false
-	if genesis.Config == nil {
-		genesis.Config = new(params.ChainConfig)
+func ValidateGenesisConfig(g *blockchain.Genesis) error {
+	if g.Config.ChainID == nil {
+		return errors.New("chainID is not specified")
 	}
 
-	// using Clique as a consensus engine
-	if genesis.Config.Istanbul == nil && genesis.Config.Clique != nil {
```

Comment:
Is this line unnecessary in the refactored version? I couldn't find this line.

Reply:
These lines match up with line 269 ~ 273 in the refactored version.

Reply:
```
if c.Governance == nil {
	engineType := c.GetConsensusEngine()
	c.Governance = GetDefaultGovernanceConfig(engineType)
	logger.Warn("Override the default governance config to the chain config", "engineType", engineType)
}
```
Do you mean above codes?

Reply:
Yes. This `if` branch and the next `else if` branch set default governance config if it doesn't exist. The above code you quoted does the same behaviors.